nixos/services/percona.nix

```diff
@@ -13,7 +13,9 @@ let
 
   mysql = cfg.package;
 
-  initFile = "/run/mysqld/init_set_root_password.sql";
+  initFile = pkgs.writeText "mysql_set_root_socket" ''
```

Comment:
As we do the transition to socket auth for _all_ supported percona/mysql versions: Let's only introduce that change into 25.05.
- needs changelog entry about root user auth changes
- upgrade path: percona83 will still be available in 25.05 as a transition path
- if this explodes in real-world staging system, reconsider this decision